File: 2022_2023/230206/김승희/PRG_순위검색_김승희.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;
class Solution {
    public int[] solution(String[] info, String[] query) {
        int[] answer = new int[query.length];
        
        for(int i = 0; i < query.length; i++){
            String[] cond = query[i].split(" ");
            int cnt = 0;
            for(int j= 0; j < info.length; j++){
                boolean chk = true;
                String[] temp = info[j].split(" ");
                int score = Integer.parseInt(temp[temp.length-1]);
                int standard = Integer.parseInt(cond[cond.length-1]);
                if(score < standard){
                    continue;
                }
                for(int k = 0, p = 0; k < cond.length-1; k++){
                    if(cond[k].equals("and")){
                        continue;
                    }
                    if(cond[k].equals("-") ){
                        p++;
                        continue;
                    }
                    boolean result = Pattern.matches(".*" + cond[k] + ".*", temp[p++]);
                    if(!result){
                        chk = false;
                        break;
                    }
                }
                
                if(chk){
                    cnt++;
                }
            }
            answer[i] = cnt;
        }
        return answer;
    }
}

File: 2022_2023/230206/김승희/PRG_순위검색_김승희.py
import re
def solution(info, query):
    answer = []
    for i in range(len(query)):
        cond = query[i].split(" ")
        cnt = 0
        for j in range(len(info)):
            chk = True;
            for k in range(len(cond)-1):
                if cond[k] == "-" or cond[k] == "and":
                    continue
                if cond[k] not in info[j]:
                    chk = False
                    break
            temp = info[j].split(" ")
            score = int(temp[len(temp)-1])
            standard = int(cond[len(cond)-1])
            if(score < standard):
                chk = False;
            if(chk):
                cnt+=1
        answer.append(cnt)
    return answer